src/script/interpreter.cpp

```diff
@@ -1303,12 +1303,12 @@ class CTransactionSignatureSerializer
         it = itBegin;
         while (scriptCode.GetOp(it, opcode)) {
             if (opcode == OP_CODESEPARATOR) {
-                s.write((char*)&itBegin[0], it-itBegin-1);
+                s.write(reinterpret_cast<const char*>(&itBegin[0]), it-itBegin-1);
```

Comment:
There is a safe helper `CharCast` and `UCharCast`, that can be used.

Though, I have a patch (Bundle 2/2 of #21969) to remove `char` completely.

Reply:
See #22167

Reply:
> See #22167


See #23438 :sweat_smile: